crypto/zkp/primitives/qrspecialrsaproofs/representation_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package qrspecialrsaproofs
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xlab-si/emmy/crypto/common"
+	"github.com/xlab-si/emmy/crypto/groups"
+)
+
+// TestQRSpecialRSA demonstrates how to prove that for a given y you know secrets x_1,...,x_k
+// such that y = g_1^x_1 * ... * g_k^x_k where g_i are given generators (bases) of QRSpecialRSA.
+func TestQRSpecialRSA(t *testing.T) {
+	group, err := groups.NewQRSpecialRSA(1024)
+	if err != nil {
+		t.Errorf("error when creating QRSpecialRSA group: %v", err)
+	}
+
+	var bases [3]*big.Int
```

Comment:
Please modify L37-L38 like this:

```go
bases := make([]*big.Int, 3)
	for i := 0; i < len(bases); i++ { ... }
```
There's no need to first define an array and use `bases[:]` as arguments to `NewRepresentationProver` and `SetProofRandomData`. I think it's better to just define a slice at the beginning and then simply pass `bases` to these functions.

The same holds for `secrets`.